File: thrifts-dotnet/ThriftS.Serializer/ListContainer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThriftS.Common;
using ThriftS.Common.Attributes;

namespace ThriftS.Serializer
{
    [ThriftSModel]
    internal class ListContainer<T>
    {
        [ThriftSMember(1)]
        public List<T> Value { get; set; }

        public ListContainer()
        {
        }

        public ListContainer(List<T> value)
        {
            Value = value;
        }
    }
}

File: thrifts-java/thrifts-serializer/src/main/java/thrifts/serializer/ListContainer.java
package thrifts.serializer;

import thrifts.common.annotations.ThriftSMember;
import thrifts.common.annotations.ThriftSModel;

import java.util.ArrayList;
import java.util.List;

@ThriftSModel
final class ListContainer<T> {

    @ThriftSMember(tag = 1)
    private ArrayList<T> value;

//    private Class<?> itemType;

    public ListContainer()
    {
    }

    public ListContainer(ArrayList<T> value)
    {
        this.value = value;
//        this.itemType = value.getClass();

    }

    public List<T> getValue() {
        return this.value;
    }

    public void setValue(ArrayList<T> value) {
        this.value = value;
    }
}
